book/Zapp-Pipes/12.-Datasource-for-rokuTV.md

```diff
@@ -0,0 +1,82 @@
+# Data Source Provided (DSP) for Roku-TV
+
+**This documentation is in beta**
+
+The Roku system cannot parse the javascript code, so the typical DSP plugin don't work for the roku TV.
+
+In order to solve this issue, we need 2 steeps :
+
+1) Modify the `plugin_manifest.json` on the DSP
+
+2) Create an external rest service for provide the data received from the client API
+
+### Prerequisites
+
+Before the next step, please make sure you are familiar with the DataSourceProvider concept, documentation [here](Home.md) .
+
+## Step 1. Update the manifest to V2
+
+Roku system parse the manifest and use the endpoint provided inside the zapp_pipes/v2 section.
+
+This is the object structure need to be added on the `plugin-manifest.json`
+
+```json
+    {
+      "zapp_pipes": {
+        "v2": {
+          "data_types": [
+            {
+              "label": "(string) Name of the feed",
+              "value": "(string) ex.: APPLICASTER_COLLECTION",
+              "endpoint": "(string) name of the endpoint",
+              "documentation": {
+                "link": "(string) link to the doc, ex: https://applicaster.zendesk.com/hc/en-us/articles/204644569-Collections",
+                "feed_id_description": "(String) Feed description For example: `accounts/xxx/broadcasters/yyy/collections/zzz`",
+                "feed_id_description_image_url": "(String) Link to image url description",
+                "feed_id_placeholder": "(String) Type the Applicaster Collection ID"
+              }
+            }
+          ],
+          "endpoints": [
+            {
+              "name": "(String) name of the endpoint: for Example : 'app'",
+              "remoteUrl": "(String) Path to the remote server, for Example: https://admin.applicaster.com/pipesv2/app/feed",
+              "contextKeys": [
+                "(String) Parameters inside the app configuration need to be sent to the external server, for Example: 'bundleIdentifier'"
+              ]
+            }
+          ]
+        }
+      }
+    }  
+```
+
+***Complete Manifest Real Example*** [Here](https://zapp.applicaster.com/admin/plugin_versions/9909/plugin_manifests/9909)
+
+#### Note about the ContextKeys :
+The `ContextKeys` array, send to the server (if needed) some parameter stored on the roku system (like authorization ID etc..) .
+
+The server receive this parameter as a normal query string, named CTX, that's base64 encoded strings.
+
+An example of query can be: "https://pipes.applicaster.com/v2/applicaster/2.0.0/live?feedId=1&ctx=#CTX"
+
+## Step 2: Create a Rest server
+
+PreRequisite : What is a rest server [here](https://www.codecademy.com/articles/what-is-rest)
+
+Each Developer can write a rest server in any language, the important thing is return, on the responses, the objects described on the applicaster feed schema API documentantion [here](https://developer.applicaster.com/Zapp-Pipes/5.-Feed-API.html).
+
+#### Node Rest Server Template 
```

Comment:
Please create a section at the same level as Step 1/2 that will contain the rest server sections.